### ISSUE TITLE
Feature Request: Latex Editor with Tabs

### ISSUE THREAD

**Issue by victordias** [sharelatex/sharelatex#482]
It would be great if the sharelatex could manage the files in tabs. It would be a great implementation and improved usability. The user could switch quickly between the files you are working on.

Something like this:
![sharelatex-with-tabs](https://cloud.githubusercontent.com/assets/1564506/19653290/bcf16202-99f2-11e6-9d7e-bb2edcef3971.png)

**Comment by henryoswald** [comment]
Thanks, this is something we will consider but it unlikely as you would be surprised how easily users can get confused by windows and tabs.

**Comment by pbarbosa** [comment]
+1

**Comment by jonatansouza** [comment]
I think it is a great idea, files in tabs would be very useful!

**Comment by allanmatheus** [comment]
Almost every text editor/IDE allow multiple tabs, including VIm. This is an expected feature (at least for me).

**Comment by henryoswald** [comment]
thanks, we have to keep in mind that people on github are power users but a lot of normal people who use ShareLaTeX have relatively basic computer skills which we are trying to help by making the editor simple. You would be surprised at how poor the level of many peoples understanding of computers is.

never say never but its not likely in the short to medium term.

**Comment by Liebeck** [comment]
You could implement this as an optional feature.

**Comment by pefi1011** [comment]
It would be great to have the option to show different .tax files in tabs or even to show different sections of a single .tex file as tabs. The option would optimize the workflow.

**Comment by Vicropht** [comment]
I must apologize for reviving this feature request, but I can only agree with the others here, tabs are one of the most common features, especially in browsers. I see no reason why this shouldn't be implemented.

**Comment by khaferkamp** [comment]
This would be big help! +1

**Comment by tcilloni** [comment]
+1 super helpful

**Comment by appukuttan-shailesh** [comment]
+1
I imagine even inexperienced computer users are now comfortable using tabs (e.g. considering how common they are in browsers).... if still in doubt, this could be an optional feature (enabled/disabled)

Will greatly enhance productivity.... esp when needing to add new references to the bib file, not having to scroll and locate the file, and then scroll again and find the tex file. Even more troublesome when handling multiple tex files.

**Comment by bhsz** [comment]
+1

**Comment by jucor** [comment]
> Will greatly enhance productivity.... esp when needing to add new references to the bib file, not having to scroll and locate the file, and then scroll again and find the tex file. Even more troublesome when handling multiple tex files.

Just ran into this exact use case from @appukuttan-shailesh.

**Comment by facundoq** [comment]
This and grammarly integration are showstoppers for me. Would definitely buy overleaf if these features were added.

**Comment by souraavv** [comment]
+1

**Comment by wltjr1007** [comment]
+1

**Comment by mlzxy** [comment]
would be a great paid feature btw

**Comment by trancelestial** [comment]
+1

**Comment by KeilSiegen** [comment]
+1

**Comment by mhadadian** [comment]
After seven years, nowadays people are much more comfortable with tabs as they are almost everywhere. Especially in browsers that they are using to open overleaf.

**Comment by AghiadHaloul** [comment]
A much, much need feature. People are far more used now to multiple tabs in multiple views in multiple windows, thanks to the increased popularity of VSCode.

**Comment by edwardyehuang** [comment]
+++++++++++++++++1

**Comment by corpaul** [comment]
I second this... this would be extremely helpful (in particular, as mentioned above, to add references to the .bib file, which is now tedious)

**Comment by samubura** [comment]
+1, yet another QoL feature that is randomly missing

**Comment by paulgavrikov** [comment]
+1 This is such an essential thing when working on long documents. For instance, switching from my main doc to insert a new bib item requires me to find the line I was working. Currently, I keep one OverLeaf tab for my bib and one for the doc, but this is less than ideal. Please, please, please, add tabs!

**Comment by matthewfeickert** [comment]
I am having a terrible time trying to collaborate on an Overleaf document that has many files that are all in play without tabs. I would normally just write my parts on my local machine with my local distribution of LaTeX and then use Git to sync up again, but in this particular document there are too many people all editing files at once (and while great that Overleaf makes such things possible!) it is a very slow and painful experience. Please add the ability to optionally enable tabs!

**Comment by crew-carson** [comment]
How is this not already the case with the file tree sidebar on the left? Are people really just asking for horizontal tabs instead of vertical tabs...

**Comment by paulgavrikov** [comment]
@crew-carson no. Loading docs from the sidebar is state less, i.e. I have to scroll to the position I was at before. Tabs should retain the state (I guess this could also work with the sidebar). Beyond that, working with docs across different folders is a nightmare

**Comment by crew-carson** [comment]
@paulgavrikov, it maintains the scroll position and even cursor position for me, but I'm new to Overleaf. Maybe there are timeouts or other limits in place. I can agree that having a recently-opened horizontal tabbar similar to Visual Studio Code seems reasonable and unlikely to confuse.

**Comment by alexandredoyen29** [comment]
+1

**Comment by phygibs** [comment]
> It would be great if the sharelatex could manage the files in tabs. It would be a great implementation and improved usability. The user could switch quickly between the files you are working on.
> 
> Something like this: ![sharelatex-with-tabs](https://cloud.githubusercontent.com/assets/1564506/19653290/bcf16202-99f2-11e6-9d7e-bb2edcef3971.png)

Still not available, for some reason! Supporting this.

**Comment by MildogMiller** [comment]
+1 I am coming up to the end of my dissertation, and during the editing/refining process, i need to jump between tabs alot. I think this feature would be an absolute game changer when it comes to working with research.

**Comment by pheeria** [comment]
In your account settings, one can join Overleaf Labs.
Editor tabs are currently one of the ongoing experiments.
